repository/repository.go

```diff
@@ -87,8 +86,9 @@ func (r *Repository) LoadAndDecrypt(t backend.Type, id backend.ID) ([]byte, erro
 }
 
 // LoadBlob tries to load and decrypt content identified by t and id from a
-// pack from the backend.
-func (r *Repository) LoadBlob(t pack.BlobType, id backend.ID) ([]byte, error) {
+// pack from the backend, the result is stored in buf, which must be large
+// enough to hold the complete blob.
+func (r *Repository) LoadBlob(t pack.BlobType, id backend.ID, buf []byte) ([]byte, error) {
```

Comment:
That's correct, but `len` of `buf` could've been modified (and this happens). A slice in Go is call-by-value regarding `len`, but call-by-reference regarding the underlying array. You can think of it as `struct {len int, cap int, data *[200]byte}`, ie. the struct is passed by value, but `data` is a pointer.

So we could either explicitely return `n` (like `Read`), or return the buf slice back to the caller.